src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1109,6 +1143,10 @@ public void setExternalFileAttributes(ZipEntry ze, int externalFileAttributes) {
     // Represents the resolved name and position of a CEN record
     static record EntryPos(String name, int pos) {}
 
+    // Implementation note: This class must be thread-safe.
+    // Each instance of Source could be shared between multiple different instances of ZipFile.
+    // Although ZipFile isn't thread-safe, the fact that separate instances of ZipFile could
+    // still concurrently use the same Source instance, mandates that Source must be thread-safe.
```

Comment:
If you assert previously that the class is thread safe then I think you can drop this confusing comment.

Reply:
Done.